app/controllers/physical_server_controller.rb

```diff
@@ -45,5 +47,15 @@ def button
 
     return if %w(physical_server_protect physical_server_tag).include?(params[:pressed]) &&
               @flash_array.nil? # Some other screen is showing, so return
+    if params[:pressed] == "#{table_name.singularize}_timeline"
+      @showtype = "timeline"
+      @record = find_record_with_rbac(ManageIQ::Providers::PhysicalInfraManager::PhysicalServer, params[:id])
+      @timeline = @timeline_filter = true
+      @lastaction = "show_timeline"
+      tl_build_timeline
+      drop_breadcrumb(:name => _("Timelines"), :url => show_link(@record, :refresh => "n", :display => "timeline"))
+      session[:tl_record_id] = @record.id
+      javascript_redirect(polymorphic_path(@record, :display => 'timeline').sub!('.', '/show/'))
```

Comment:
@lucashsilva Since you have defined the mixin `Mixins::MoreShowActions` at the top, could you simply call the method `show_timeline` instead of writing this block?

Reply:
@AparnaKarve I figured out I didn't use anything from this mixin and that the model I followed didn't either. Sorry for that. I also tried to do what you said but the user is not redirected to the timelines page and nothing is shown. If you look at [this](https://github.com/ManageIQ/manageiq-ui-classic/blob/master/app/controllers/ems_common.rb#L289), you'll see a similar code, which was the way I found to redirect the user to the timelines page. How should I proceed?

Is it necessary to execute the lines 51 to 57 if the user will be redirected to a new page? 
I could get the button to work and redirect correctly with lines 52 and 58 only and without the mixin.

Reply:
@lucashsilva I think it would be better to use the mixin.
The eventual goal would be to make use of the mixin in the entire app and remove duplication.

Can you give this a try -
```ruby
if params[:pressed] == "physical_server_timeline"
  @record = find_record_with_rbac(ManageIQ::Providers::PhysicalInfraManager::PhysicalServer, params[:id])
  show_timeline
  javascript_redirect :action => 'show', :id => @record.id, :display => 'timeline'
end
```

Reply:
@AparnaKarve I tried it and it works. I already pushed the change.